src/main/java/seedu/address/logic/commands/BookmarkCommand.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.company.Bookmark;
+import seedu.address.model.company.Company;
+
+/**
+ * Bookmarks a company identified using its displayed index from the address book.
+ */
+public class BookmarkCommand extends Command {
+
+    public static final String COMMAND_WORD = "bookmark";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": bookmarks the company based on the index"
+            + "in the list of contacts.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 2";
+
+    public static final String MESSAGE_BOOKMARK_SUCCESS = "Bookmarked company: %1$s";
+
+    public static final String MESSAGE_BOOKMARK_FAILURE = "Company is already bookmarked: %1$s";
+
+    private final Index index;
+
+    /**
+     * Constructor for the BookmarkCommand
+     *
+     * @param index The index of the company to be bookmarked in the address book.
+     */
+    public BookmarkCommand(Index index) {
+        requireAllNonNull(index);
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Company> lastShownList = model.getFilteredCompanyList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
```

Comment:
Checking for index out of bounds condition is good here, shows good defensive programming